src/Experimental/ShortestPaths/ShortestPaths.jl

```diff
@@ -0,0 +1,105 @@
+module ShortestPaths
+using SparseArrays: sparse
+using LightGraphs
+using LightGraphs:AbstractPathState, DijkstraState, BellmanFordState,
+      FloydWarshallState, DEsopoPapeState, JohnsonState, AbstractGraph, AbstractEdge
+#
+# using LightGraphs: dijkstra_shortest_paths, bellman_ford_shortest_paths, floyd_warshall_shortest_paths,
+#     desopo_pape_shortest_paths, a_star, johnson_shortest_paths
+import Base:convert, getproperty
+import LightGraphs: enumerate_paths
+
+using DataStructures:PriorityQueue, enqueue!, dequeue!
+
+
+struct LGEnvironment
+    threaded::Bool
+    parallel::Bool
+    LGEnvironment() = new(false, false)
+end
```

Comment:
In C++ we have something similar called an exectution policy: https://en.cppreference.com/w/cpp/algorithm/execution_policy_tag_t.

Maybe we could also adapt this name.

Maybe we can also work with `ExecutionPolicy` as abstract type, and then have subtypes such as `GpuPolicy`, `ThreadPolicy`, `MultiCorePolicy`. They could also take some potential parameteres, such as max num of threads.

Reply:
Sounds awesome! I'm going to remove this for now though since we're not using it.